lib/ClangImporter/ClangImporter.cpp

```diff
@@ -829,7 +829,7 @@ bool ClangImporter::canReadPCH(StringRef PCHFilename) {
   auto result = Reader->ReadAST(PCHFilename,
                   clang::serialization::MK_PCH,
                   clang::SourceLocation(),
-                  clang::ASTReader::ARR_None);
+                  clang::ASTReader::ARR_OutOfDate);
```

Comment:
Oh. Seriously? Wow.

We should probably add all of them here, or at least all but ARR_ConfigurationMismatch, since we also handle VersionMismatch by rebuilding and we *could* handle Missing here too (but don't need to because we checked it above).

Reply:
Right. I've added ARR_Missing and ARR_VersionMismatch, so all expected errors should be listed.